TimeService/work/timeservice/implem/default/C/src/timeservice-api.c

```diff
@@ -0,0 +1,94 @@
+#include "timeservice-api.h"
+
+#define BYTE_CAPACITY 256
+#define CUC_TFIELD_SIZE 7
+
+asn1SccCucTimestamp nanoseconds_to_cuc_timestamp(uint64_t nanoseconds)
+{
+    asn1SccCucTimestamp timestamp;
+    long double seconds = (long double)nanoseconds / NS_PER_SECOND;
+
+    // CUC timestamp contains three bytes for fractions of a second
+    uint64_t seconds_shifted = seconds * BYTE_CAPACITY * BYTE_CAPACITY * BYTE_CAPACITY; 
+    seconds_shifted = adjust_endianess(seconds_shifted);
+    unsigned char *time = (unsigned char *)&seconds_shifted;
+
+    for (size_t i = 0; i < CUC_TFIELD_SIZE; i++)
+    {
+        timestamp.arr[i] = time[CUC_TFIELD_SIZE - 1 - i];
+    }
+    return timestamp;
+}
+
+void TimeService_Startup(TimeService *const self)
+{
+    self->starting_time = get_time();
+}
+
+asn1SccComparisonResult TimeService_CfsTimestampCmp(TimeService *const self, const asn1SccCfsTimestamp *timestamp1, const asn1SccCfsTimestamp *timestamp2)
+{
+    // Unused in this implementation
+    (void)self;
+
+    if (timestamp1->seconds > timestamp2->seconds)
+    {
+        return 1;
+    }
+    else if (timestamp1->seconds < timestamp2->seconds)
+    {
+        return -1;
+    }
+    else if (timestamp1->seconds == timestamp2->seconds)
```

Comment:
After two previous ifs, is this comparison needed?

Reply:
No, removed